src/lib/events.tsx

```diff
@@ -0,0 +1,97 @@
+import { ENV } from '../lib/env'
+
+const LOCAL_USER_ID_FUNCTION_HANDLER = 'local_fe_events_direct_user_id_handler'
+const LOCAL_FE_EVENTS_MOODLE_FUNCTION_HANDLER = 'local_fe_events_moodle_content_loaded_handler'
+const LOCAL_FE_EVENTS_DIRECT_FUNCTION_HANDLER = 'local_fe_events_direct_content_loaded_handler'
+
+const PIPELINE_MOODLE_EVENTS = 'moodle_eventing'
+const PIPELINE_MOODLE_DIRECT = 'moodle_direct'
+const PIPELINE_FE_DIRECT = 'fe_direct'
+
+function getMoodlePluginMethodUrl(endpointMethod: string): string {
+  if (window.M === undefined) {
+    console.log('moodle variables not found')
+    return ''
+  }
+  let userUrl: string
+  userUrl = String(window.M.cfg.wwwroot)
+  userUrl += '/lib/ajax/service.php?sesskey='
+  userUrl += String(window.M.cfg.sesskey)
+  userUrl += '&info=' + endpointMethod
+  return userUrl
+}
+
+async function sendEventMoodlePlugin(pipelineId: string): Promise<void> {
+  const contentId = 'moodle_test_content_via_moodle_direct'
```

Comment:
See related content, but let's replace this placeholder with the actual `data-content-id` value.